File: backend-2024-grupo2/src/main/kotlin/ar/edu/unsam/phm/backendgrupo2/domain/StatusShow.kt
package ar.edu.unsam.phm.backendgrupo2.domain

import ar.edu.unsam.phm.backendgrupo2.dto.StatusShowDTO
import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "status_show")
class StatusShow {
  
  companion object {
    
    fun fromJSON(statusShowDTO: StatusShowDTO): StatusShow =
      StatusShow().apply {
        this.id = statusShowDTO.id
        this.name = statusShowDTO.name
        this.profitStatusShowType = statusShowDTO.profitStatusShowType.toBigDecimal()
      }
  }
  
  @Id
  @GeneratedValue
  var id: Int = 0
  
  @Column(length = 40)
  var name: String = ""
  
  @Column
  var profitStatusShowType: BigDecimal = 0.toBigDecimal()
  
  fun profit(): BigDecimal =
    this.profitStatusShowType
}

File: frontend-2024-grupo2/src/app/data/models/StatusShow.ts
export type StatusShowJSON = {
  name: string
  profitStatusShowType: number
}

export class StatusShow {
  constructor(
    public name: string,
    public profitStatusShowType: number
  ) {}

  static fromJSON(statusShowJSON: StatusShowJSON): StatusShow {
    return Object.assign(
      new StatusShow(statusShowJSON.name, statusShowJSON.profitStatusShowType)
    )
  }

  toJSON(): StatusShowJSON {
    return {
      name: this.name,
      profitStatusShowType: this.profitStatusShowType
    }
  }
}
